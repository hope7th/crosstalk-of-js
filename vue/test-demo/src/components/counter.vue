<template>
    <div>
        <span>{{count}}</span>
        <button @click="handleclick">count++</button>
    </div>
</template>

<script>
    export default {
        name: "counter",
        data(){
            return {
                count:0
            }
        },
        methods:{
            handleclick(){
                this.count++;
                this.$emit("change",this.count)
            }
        }
    }
</script>

<style scoped>

</style>