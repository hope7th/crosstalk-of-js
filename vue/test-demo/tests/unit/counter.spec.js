import { mount } from "@vue/test-utils";
import counter from "@/components/counter.vue";
import sinon from 'sinon'

describe("counter.vue", () => {
  let isCalled = false;
  const change = sinon.spy();
  const wrapper = mount(counter,{
  listeners:{
    change
  }
  });
  it("renders couter html", () => {
    const msg = "new message";
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("counter++",()=>{
    const button = wrapper.find("button");
    button.trigger("click");
    expect(wrapper.vm.count).toBe(1);
    expect(change.called).toBe(true);
  })
});
