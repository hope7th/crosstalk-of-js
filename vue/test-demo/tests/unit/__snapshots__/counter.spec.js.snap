// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter.vue renders couter html 1`] = `<div><span>0</span> <button>count++</button></div>`;

exports[`counter.vue renders props.msg when passed 1`] = `<div><span>0</span> <button>count++</button></div>`;
